### PR TITLE
Workaround plotly / kaleido static figure generation

### LINKED ISSUE #1089: uni-tests taking for ever
## **Motivation**

Reduce the time required to run the uni-tests.

## **Cause**

Related to figure generation

### PR DESCRIPTION
**Pull request by VGPReys**
## Checklist

- [X] Tests added for the new code
- [ ] Documentation added for the code changes
- [ ] Does not break licensing
- [X] Does not add any dependencies, if it does please add a thorough explanation

## Summary of the Pull Request

- Removed the `kaleido` package for mandatory packages
- Added `kaleido`as optional package
- Do not generate matrix plots (in `[clustfcc]` and `[clustrmsd]`) when over 3100 entries (3100 * 3100 points).
- Added uni-tests for the generation of plots performed by `haddock3-analyse --format png`
- Added checks in `haddock3-analyse` when `--format` option is specified to make sure `kaleido` is installed.

## Related Issue

Closes  #1089

### CODE REVIEW

File: src/haddock/libs/libfunc.py [hunk 1]
```diff
@@ -83,3 +86,15 @@ def nan(*ignore: Any, **everything: Any) -> float:
 def not_none(value: Any) -> bool:
     """Give True if value is not None, or False otherwise."""
     return is_not(value, None)
+
+
+def exec_time(func):
```
Comment (rvhonorato) [inline src/haddock/libs/libfunc.py:91]: nice!

File: tests/test_libplots.py [hunk 2]
```diff
@@ -262,3 +287,67 @@ def test_offline_js_manager():
     plotly_cdn_full_url = plotly_cdn_url()
     offline_plotly_js = offline_js_manager(tmpdir, offline=False)
     assert plotly_cdn_full_url in offline_plotly_js
+
+
+def test_box_plot_handler(example_capri_ss, cluster_ranking):
+    """Test box plot generation without format definition."""
+    initdir = os.getcwd()
+    with tempfile.TemporaryDirectory() as tmpdir:
+        os.chdir(tmpdir)
```
Comment (rvhonorato) [inline tests/test_libplots.py:296]: Not sure how these functions work, but can you avoid this (and the others) `chdir` or do they expect the files to be relative?

File: src/haddock/clis/cli_analyse.py [hunk 3]
```diff
@@ -491,11 +507,58 @@ def analyse_step(
...
+    if format in SUPPORTED_OUTPUT_FORMATS:
+        # Make sure the `kaleido` package is installed
+        try:
+            import kaleido  # noqa : F401
+        except ImportError:
+            raise ImportError(
+                f"Exporting with format {format} requires the use of `kaleido`"
+                f" package, that seems not to be installed. {os.linesep}"
+                "Please install it with: `pip install kaleido==0.2.*`"
+                )
```
Comment (rvhonorato) [inline src/haddock/clis/cli_analyse.py:550]: How is `kaleido` affecting the time the tests take to run?

File: tests/test_libplots.py (hunk 2 above)
Comment (VGPReys) [reply tests/test_libplots.py:296]: - relative files
- easier to delete everything (will generate bunch of `.html` and `.png` files...)

File: src/haddock/clis/cli_analyse.py (hunk 3 above)
Comment (VGPReys) [reply src/haddock/clis/cli_analyse.py:550]: `kaleido` is used to generate `.png` files from plotly figures.
It takes quite some time.
Especially when generating figures with millions of data points.

**Review by rvhonorato — APPROVED** [review]

**Pull request merged**